docs/sources/reference/commandline/cli.md

```diff
@@ -533,6 +533,10 @@ considered as normal files. Filepaths in .dockerignore are absolute with
 the current directory as the root. Wildcards are allowed but the search
 is not recursive.
 
+When the `URL` references a Dockerfile, or a Dockerfile is read from 
+`STDIN`, then the `-f` option will have no effect and the contents of the
+Dockerfile will be placed in a file called `Dockerfile`.
+
```

Comment:
I'd move this up in the description content.  Also, the limitation on the `-f` flag should go with that flag's description.


-----

Builds Docker images from a Dockerfile and a "context". A build's context are the files located in the specified `PATH` or `URL`.  The build process can refer to any of the files in the context. For example, your build can use an [*ADD*](/reference/builder/#add) instruction to reference a file in the context.

The `URL` parameter can specify the location of a Git repository; in this case,  the repository is the context. The Git repository is recursively cloned with its submodules.  The system does a fresh `git clone -recursive` in a temporary directory on your local host. Then, this clone is sent to the Docker daemon as the context. Local clones give you the ability to access private repositories using local user credentials, VPN's, and so forth.

Instead of specifying a context, you can pass a single Dockerfile in the `URL` or pipe the file in via `STDIN`.  To pipe a Dockerfile from `STDIN`:

	docker build - < Dockerfile

If you use STDIN or specify a `URL`, the system places the contents into a file called `Dockerfile`. In this scenario, there is no context.

If a file named `.dockerignore` exists in the root of `PATH` then it
is interpreted as a newline-separated list of exclusion patterns.
Exclusion patterns match files or directories relative to `PATH` that
will be excluded from the context. Globbing is done using Go's
[filepath.Match](http://golang.org/pkg/path/filepath#Match) rules.

Please note that `.dockerignore` files in other subdirectories are
considered as normal files. Filepaths in `.dockerignore` are absolute with
the current directory as the root. Wildcards are allowed but the search
is not recursive.

----

 -f, --file=""            Name of the Dockerfile(Default is 'Dockerfile'). This is ignored if a Dockerfile is supplied via a `URL` or `STDIN`.